tests/test_new_return_types.py

```diff
@@ -1174,6 +1175,975 @@ def circuit(x):
+        assert res.shape == (2, 3)
+
+    @pytest.mark.torch
+    def test_multiple_expval_torch(self):
+        """Return Jacobian of multiple expvals."""
+        import torch
+
+        dev = qml.device("default.qubit", wires=2)
+        qml.enable_return()
+
+        @qml.qnode(dev, interface="torch")
+        def circuit(a):
+            qml.RX(a[0], wires=0)
+            qml.CNOT(wires=(0, 1))
+            qml.RY(a[1], wires=1)
+            qml.RZ(a[2], wires=1)
+            return qml.expval(qml.PauliZ(wires=0)), qml.expval(qml.PauliZ(wires=1))
+
+        x = torch.tensor([0.1, 0.2, 0.3])
+
+        res = torch.autograd.functional.jacobian(circuit, x)
+
+        qml.disable_return()
+
+        assert isinstance(res, tuple)
+        assert len(res) == 2
+        for elem in res:
+            assert isinstance(elem, torch.Tensor)
+            assert elem.shape == (3,)
+
+    @pytest.mark.tf
+    def test_multiple_expval_tf(self):
+        """Return Jacobian of multiple expvals."""
+        import tensorflow as tf
+
+        dev = qml.device("default.qubit", wires=2)
+        qml.enable_return()
+
+        @qml.qnode(dev, interface="tf")
+        def circuit(a):
+            qml.RX(a[0], wires=0)
+            qml.CNOT(wires=(0, 1))
+            qml.RY(a[1], wires=1)
+            qml.RZ(a[2], wires=1)
+            return qml.expval(qml.PauliZ(wires=0)), qml.expval(qml.PauliZ(wires=1))
+
+        x = tf.Variable([0.1, 0.2, 0.3])
+
+        with tf.GradientTape() as tape:
+            out = circuit(x)
+
+        res = tape.jacobian(out, x)
+
+        qml.disable_return()
+
+        assert isinstance(res, tf.Tensor)
+        assert res.shape == (2, 3)
+
+    @pytest.mark.jax
+    def test_multiple_expval_jax(self):
+        """Return Jacobian of multiple expvals."""
+        import jax
+
+        dev = qml.device("default.qubit", wires=2)
+        qml.enable_return()
+
+        @qml.qnode(dev, interface="jax")
+        def circuit(a):
+            qml.RX(a[0], wires=0)
+            qml.CNOT(wires=(0, 1))
+            qml.RY(a[1], wires=1)
+            qml.RZ(a[2], wires=1)
+            return qml.expval(qml.PauliZ(wires=0)), qml.expval(qml.PauliZ(wires=1))
+
+        x = jax.numpy.array([0.1, 0.2, 0.3])
+        res = jax.jacobian(circuit)(x)
+
+        qml.disable_return()
+
+        assert isinstance(res, tuple)
+        assert len(res) == 2
+        for elem in res:
+            assert isinstance(elem, jax.numpy.ndarray)
+            assert elem.shape == (3,)
+
+    @pytest.mark.autograd
+    def test_multiple_probs_autograd(self):
+        """Return Jacobian of multiple probs."""
+        dev = qml.device("default.qubit", wires=2)
+        qml.enable_return()
+
+        @qml.qnode(dev)
+        def circuit(a):
+            qml.RX(a[0], wires=0)
+            qml.CNOT(wires=(0, 1))
+            qml.RY(a[1], wires=1)
+            qml.RZ(a[2], wires=1)
+            return qml.probs(op=qml.PauliZ(wires=0)), qml.probs(wires=1)
```

Comment:
Could we have the second probs apply to more than one wire?

That case created a ragged array previously.